### PR TITLE
Update vendor files for backupstore.

### PR DESCRIPTION
**Pull request by james-munson**
https://github.com/longhorn/longhorn/issues/6608

Accept NFS mount options as a query appended to the backup target URL.

This PR is a placeholder for the vendor update to be done after backupstore is merged.

### CODE REVIEW

File: vendor/github.com/longhorn/backupstore/deltablock.go [hunk 1]
```diff
@@ -1100,6 +1101,7 @@ func getLatestBackup(backup *Backup, lastBackup *Backup) error {
 	}
 
 	if backupTime.After(lastBackupTime) {
+		// FIXME - go lint points out that this copies a potentially locked sync.mutex
```
Comment (PhanLe1010) [inline vendor/github.com/longhorn/backupstore/deltablock.go:1104]: is this comment intended 
```
// FIXME - go lint points out that this copies a potentially locked sync.mutex
```

**Review by PhanLe1010 — COMMENTED** [review]
In gereral LGTM. A small question

File: vendor/github.com/longhorn/backupstore/deltablock.go (hunk 1 above)
Comment (james-munson) [reply vendor/github.com/longhorn/backupstore/deltablock.go:1104]: Yes.  It's a lint error that I got while running :GoLint in my IDE, and it looks real, but not to address in this PR.

**Review by PhanLe1010 — APPROVED** [review]

**Comment by james-munson** [comment]
@mergify backport v1.5.x

**Pull request merged**